gluster-prometheus.spec

```diff
@@ -0,0 +1,126 @@
+
+%if 0%{?with_debug}
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%{!?go_arches: %global go_arches x86_64 aarch64 ppc64le }
+
+%global provider github
+%global provider_tld com
+%global project gluster
+%global repo gluster-prometheus
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path %{provider_prefix}
+%global gluster_mgmt %{getenv:GLUSTER_MGMT}
+
+%global gluster_prom_make %{__make} PREFIX=%{_prefix} EXEC_PREFIX=%{_exec_prefix} BINDIR=%{_bindir} SBINDIR=%{_sbindir} DATADIR=%{_datadir} LOCALSTATEDIR=%{_sharedstatedir} LOGDIR=%{_localstatedir}/log SYSCONFDIR=%{_sysconfdir} FASTBUILD=off
+
+%global gluster_prom_ver 1.0.0
+%global gluster_prom_rel rc0
+
+Name: %{repo}
+Version: %{gluster_prom_ver}
+Release: %{gluster_prom_rel}%{?dist}
+Summary: The GlusterFS prometheus metrics collectors
+License: GPLv2 or LGPLv3+
+URL: https://%{provider_prefix}
+%if 0%{?with_bundled}
+Source0: https://%{provider_prefix}/releases/download/v%{version}/%{name}-v%{gluster_prom_ver}-%{gluster_prom_rel}-vendor.tar.xz
+%else
+Source0: https://%{provider_prefix}/releases/download/v%{version}/%{name}-v%{gluster_prom_ver}-%{gluster_prom_rel}.tar.xz
+%endif
+
+ExclusiveArch: %{go_arches}
+
+BuildRequires: %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires: systemd
+
+%if ! 0%{?with_bundled}
+BuildRequires: golang(github.com/asaskevich/govalidator)
+BuildRequires: golang(github.com/cespare/xxhash)
+BuildRequires: golang(github.com/cockroachdb/cmux)
+BuildRequires: golang(github.com/coreos/etcd/clientv3)
+BuildRequires: golang(github.com/coreos/etcd/clientv3/concurrency)
+BuildRequires: golang(github.com/coreos/etcd/clientv3/namespace)
+BuildRequires: golang(github.com/coreos/etcd/embed)
+BuildRequires: golang(github.com/coreos/etcd/etcdserver/etcdserverpb)
+BuildRequires: golang(github.com/coreos/etcd/pkg/transport)
+BuildRequires: golang(github.com/coreos/etcd/pkg/types)
+BuildRequires: golang(github.com/coreos/pkg/capnslog)
+BuildRequires: golang(github.com/dgrijalva/jwt-go)
+BuildRequires: golang(github.com/golang/protobuf/proto)
+BuildRequires: golang(github.com/gorilla/handlers)
+BuildRequires: golang(github.com/gorilla/mux)
+BuildRequires: golang(github.com/justinas/alice)
+BuildRequires: golang(github.com/olekukonko/tablewriter)
+BuildRequires: golang(github.com/pborman/uuid)
+BuildRequires: golang(github.com/pelletier/go-toml)
+BuildRequires: golang(github.com/rasky/go-xdr/xdr2)
+BuildRequires: golang(github.com/sirupsen/logrus)
+BuildRequires: golang(github.com/spf13/cobra)
+BuildRequires: golang(github.com/spf13/pflag)
+BuildRequires: golang(github.com/spf13/viper)
+BuildRequires: golang(github.com/thejerf/suture)
+BuildRequires: golang(golang.org/x/net/context)
+BuildRequires: golang(golang.org/x/sys/unix)
+BuildRequires: golang(google.golang.org/grpc)
+BuildRequires: golang(go.opencensus.io)
+%endif
+
+Requires: glusterfs-server >= 5.0.0
```

Comment:
Are we using any features which are not available in older versions? `>= 3.12` may be good

Reply:
Ack. I dont think anything specific we are using which doesnt work with older version.